python/ray/serve/_private/logging_utils.py

```diff
@@ -47,16 +49,38 @@ class ServeJSONFormatter(logging.Formatter):
     based on the field of record.
     """
 
+    ADD_IF_EXIST_FIELDS = [
+        SERVE_LOG_REQUEST_ID,
+        SERVE_LOG_ROUTE,
+        SERVE_LOG_APPLICATION,
+    ]
+
     def __init__(
         self,
         component_name: str,
         component_id: str,
         component_type: Optional[ServeComponentType] = None,
     ):
+        runtime_context = ray.get_runtime_context()
         self.component_log_fmt = {
             SERVE_LOG_LEVEL_NAME: SERVE_LOG_RECORD_FORMAT[SERVE_LOG_LEVEL_NAME],
             SERVE_LOG_TIME: SERVE_LOG_RECORD_FORMAT[SERVE_LOG_TIME],
+            SERVE_LOG_ACTOR_ID: runtime_context.get_actor_id(),
+            SERVE_LOG_WORKER_ID: runtime_context.get_worker_id(),
         }
+        try:
+            runtime_context = ray.get_runtime_context()
+            actor_id = runtime_context.get_actor_id()
+            if actor_id:
+                self.component_log_fmt[SERVE_LOG_ACTOR_ID] = actor_id
+            worker_id = runtime_context.get_worker_id()
+            if worker_id:
+                self.component_log_fmt[SERVE_LOG_WORKER_ID] = worker_id
+        except Exception:
+            # If get_runtime_context() fails for any reason, do nothing (no adding
+            # actor_id and/or worker_id to the fmt)
+            pass
```

Comment:
I think you probably meant to put the `try-except` block _above_ the dict construction :)

Reply:
oh, I meant to remove them in line 68-69 and just have they added to the dictionary in the try-except block 😅